test/utils/noderesourcetopologies/noderesourcetopologies.go

```diff
@@ -321,15 +322,30 @@ func AccumulateNames(nrts []nrtv1alpha2.NodeResourceTopology) sets.String {
 	return nodeNames
 }
 
-func FilterTopologyManagerPolicy(nrts []nrtv1alpha2.NodeResourceTopology, tmPolicy nrtv1alpha2.TopologyManagerPolicy) []nrtv1alpha2.NodeResourceTopology {
+func FilterByTopologyManagerPolicy(nrts []nrtv1alpha2.NodeResourceTopology, tmPolicy string) []nrtv1alpha2.NodeResourceTopology {
+	return filterByAttribute(nrts, nrtv1alpha2.AttributeInfo{Name: e2enrt.TopologyManagerPolicyAttribute, Value: tmPolicy})
+}
+
+func FilterByTopologyManagerScope(nrts []nrtv1alpha2.NodeResourceTopology, tmScope string) []nrtv1alpha2.NodeResourceTopology {
+	return filterByAttribute(nrts, nrtv1alpha2.AttributeInfo{Name: e2enrt.TopologyManagerScopeAttribute, Value: tmScope})
+}
+
+func FilterByTopologyManagerPolicyAndScope(nrts []nrtv1alpha2.NodeResourceTopology, tmPolicy, tmScope string) []nrtv1alpha2.NodeResourceTopology {
+	return FilterByTopologyManagerScope(FilterByTopologyManagerPolicy(nrts, tmPolicy), tmScope)
+}
+
+func filterByAttribute(nrts []nrtv1alpha2.NodeResourceTopology, attrInfo nrtv1alpha2.AttributeInfo) []nrtv1alpha2.NodeResourceTopology {
 	ret := []nrtv1alpha2.NodeResourceTopology{}
 	for _, nrt := range nrts {
-		if !contains(nrt.TopologyPolicies, string(tmPolicy)) {
-			klog.Warningf("SKIP: node %q doesn't support topology manager policy %q", nrt.Name, string(tmPolicy))
+		attr, ok := attribute.Get(nrt.Attributes, attrInfo.Name)
+		if !ok {
+			klog.Error("SKIP: node %q doesn't have required attribute %q", nrt.Name, attrInfo.Name)
+			continue
+		}
+		if attr.Value != attrInfo.Value {
+			klog.Warningf("SKIP: node %q doesn't wrong attribute value actual:%q, expected: %q", nrt.Name, attr.Value, attrInfo.Value)
```

Comment:
this is nice, but now I wonder if we should just remove the one-line wrappers and use this everywhere

Reply:
Sure we can but I think this is much more readable
```
FilterByTopologyManagerPolicy(nrtList.Items, intnrt.SingleNUMANode)
```
than this
```
filterByAttribute(nrtList.Items, nrtv1alpha2.AttributeInfo{Name: e2enrt.TopologyManagerPolicyAttribute, Value: intnrt.SingleNUMANode})

```

And I am pretty sure that the go compiler would be able to optimize the call avoiding unnecessary jumps

Reply:
oneliner-functions which just translate/reshuffle/repack parameters are almost always a smell. In this case I think practicality trumps so let's ignore the smell and go with the current approach.

Reply:
Well, I am not sure that is a code smell ... I mean that kind of functions could make the code more readable hiding implementation details or "ugly" code compositions. 
In this case the "user" just want to filter a list by a value I do not need to know if that value is an attribute, nor need to build an AttributeInfo struct ... those are implementation details.

Reply:
I agree in this case is beneficial and improves readability in the calling site (not so sold in the implementation site, but it's good enough).
In general I've become very wary of shallow functions, which trivial-oneliners which repack/reshuffle/rearrange arguments are a prime example of.